app/stac_api/migrations/0024_added_update_interval_field.py

```diff
@@ -0,0 +1,132 @@
+        ),
+        migrations.AddIndex(
+            model_name='item',
+            index=models.Index(fields=['update_interval'], name='item_update_interval_idx'),
+        ),
+        pgtrigger.migrations.AddTrigger(
+            model_name='asset',
+            trigger=pgtrigger.compiler.Trigger(
+                name='add_del_asset_item_update_interval_trigger',
+                sql=pgtrigger.compiler.UpsertTriggerSql(
+                    declare=
+                    'DECLARE asset_instance stac_api_asset%ROWTYPE; item_update_interval RECORD;',
+                    func=
+                    "\n        asset_instance = COALESCE(NEW, OLD);\n\n        -- Compute item update_interval (minimum aggregation of asset's update_interval)\n        SELECT\n            COALESCE(MIN(NULLIF(asset.update_interval, -1)), -1) AS min_update_interval\n        INTO item_update_interval\n        FROM stac_api_asset AS asset\n        WHERE asset.item_id = asset_instance.item_id;\n\n        -- Update related item update_interval variables\n        UPDATE stac_api_item SET\n            update_interval = item_update_interval.min_update_interval\n        WHERE id = asset_instance.item_id;\n\n        RAISE INFO 'item.id=% update_interval updated, due to asset.name=% updates.',\n            asset_instance.item_id, asset_instance.name;\n\n        RETURN asset_instance;\n        ",
+                    hash='4489101822128d8d934da4095ec7dc7ec519e513',
+                    operation='INSERT OR DELETE',
+                    pgid='pgtrigger_add_del_asset_item_update_interval_trigger_60f97',
+                    table='stac_api_asset',
+                    when='AFTER'
+                )
+            ),
+        ),
+        pgtrigger.migrations.AddTrigger(
+            model_name='asset',
+            trigger=pgtrigger.compiler.Trigger(
+                name='update_asset_item_update_interval_trigger',
+                sql=pgtrigger.compiler.UpsertTriggerSql(
+                    condition='WHEN (OLD.* IS DISTINCT FROM NEW.*)',
+                    declare=
+                    'DECLARE asset_instance stac_api_asset%ROWTYPE; item_update_interval RECORD;',
+                    func=
+                    "\n        asset_instance = COALESCE(NEW, OLD);\n\n        -- Compute item update_interval (minimum aggregation of asset's update_interval)\n        SELECT\n            COALESCE(MIN(NULLIF(asset.update_interval, -1)), -1) AS min_update_interval\n        INTO item_update_interval\n        FROM stac_api_asset AS asset\n        WHERE asset.item_id = asset_instance.item_id;\n\n        -- Update related item update_interval variables\n        UPDATE stac_api_item SET\n            update_interval = item_update_interval.min_update_interval\n        WHERE id = asset_instance.item_id;\n\n        RAISE INFO 'item.id=% update_interval updated, due to asset.name=% updates.',\n            asset_instance.item_id, asset_instance.name;\n\n        RETURN asset_instance;\n        ",
```

Comment:
My pgtrigger/SQL foo still needs improvements, but if I understand this correctly, we make sure here, that the minimum will be calculated excluding the -1 values, right? Because otherwise of course we'd have -1 as a minimum, if that value appears somewhere. Which we would not want.

Reply:
Correct